### ISSUE TITLE
Streamline exposed module structure

### ISSUE THREAD

**Issue by knoellle** [Adanos020/egui_dock#61]
Currently the import section of `lib.rs` looks like this:
```rust
use egui::style::Margin;
use egui::*;

#[allow(deprecated)]
pub use crate::{
    dynamic_tab::{DynamicTabViewer, DynamicTree, Tab, TabBuilder},
    style::{Style, StyleBuilder, TabAddAlign},
    tree::{Node, NodeIndex, Split, TabIndex, Tree},
};
pub use egui;
use utils::*;

mod dynamic_tab;
mod style;
mod tree;
mod utils;
```

Every type from the sub modules like tree, style, etc. has to be manually added to the `pub use` section to be accessible to users of the crate. It essentially flattens the structure as viewed from the outside, since every type has to be imported directly from `egui_dock`.
This approach and has to be maintained manually and as such is prone to errors (see #60).

My suggestion would be to expose the sub modules directly via `pub mod` and then just non-pub using the types for use within the crate's source files to avoid long qualified paths.

This would be a breaking change, since users would have to adjust their include paths from e.g. `egui_dock::Node` to `egui_dock::tree::Node`.

**Comment by Adanos020** [comment]
I think we could make the modules public without hiding the `use`s. That would ensure that all public stuff is accessible and also won't break any more code.

**Comment by msheldyakov** [comment]
@Adanos020 i will try to do something. If you have a couple of useful links/examples of desired layout - I will be glad.
(now I'm not sure I understood everything)

**Comment by Adanos020** [comment]
@msheldyakov This is about making all modules public, except for `utils`, nothing else

**Comment by Adanos020** [comment]
Fixed in [120ce1e](https://github.com/Adanos020/egui_dock/commit/120ce1e1e65420ac8e97c11326bec7e5d6544045)
